src/pages/Accounts/AccountDetailsPage.vue

```diff
@@ -0,0 +1,460 @@
+<script setup lang="ts">
+import { useRouter } from 'vue-router';
+import { computed, onMounted, ref } from 'vue';
+import { http } from '@/shared/api';
+import BaseContentBlock from '@/shared/ui/components/BaseContentBlock.vue';
+import DataField from '@/shared/ui/components/DataField.vue';
+import { useTable } from '@/shared/lib/table';
+import { formatMoney } from '@/shared/utils/money-formatters';
+import BaseTable from '@/shared/ui/components/BaseTable.vue';
+import {
+  type filterTransactionsModel as ftm,
+  TransactionStatusFilter,
+  TransactionTypeFilter,
+} from '@/features/filter-transactions';
+import BaseHash from '@/shared/ui/components/BaseHash.vue';
+import TransactionStatus from '@/entities/transaction/TransactionStatus.vue';
+import { transactionModel } from '@/entities/transaction';
+import { useWindowSize } from '@vueuse/core';
+import { format } from '@/shared/lib/time';
+import { adaptiveTransactionTypeOptions } from './consts';
+import BaseLoading from '@/shared/ui/components/BaseLoading.vue';
+import { useErrorHandlers } from '@/shared/ui/composables/useErrorHandlers';
+import { applicationCurrency } from '@/shared/config';
+
+const router = useRouter();
+
+const { handleUnknownError } = useErrorHandlers();
+
+const HASH_BREAKPOINT = 1200;
+const { width } = useWindowSize();
+
+const hashType = computed(() => (width.value < HASH_BREAKPOINT ? 'short' : 'medium'));
+
+const accountId = computed(() => String(router.currentRoute.value.params['id']));
```

Comment:
Isn't `params['id']` already a string? If you added `String()` to convert possible `string[]` to a string, it seems to be not a desirable logic.

Consider adding simple assertions using something like [`tiny-invariant`](https://www.npmjs.com/package/tiny-invariant):

```ts
const id = router.currentRoute.value.params['id']
invariant(typeof id === 'string')
return id // TS narrowed the type
```

Reply:
Maybe just leave as the following? There won't be cases with array values
```
const id = router.currentRoute.value.params['id']
```

Reply:
Assertions generally are helpful in future. Up to you.

Reply:
According to https://stackoverflow.com/a/72839752 seems like we won't have any cases with array values but I will add an assertion anyway👍